tests/test_index.py

```diff
@@ -0,0 +1,130 @@
+#
+# Pyserini: Reproducible IR research with sparse and dense representations
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import os
+import faiss
+import unittest
+import pathlib as pl
+
+
+from pyserini.util import get_cache_home
+
+
+class TestSearch(unittest.TestCase):
+    def setUp(self):
+        self.docids = []
+        self.texts = []
+        self.test_file = 'tests/resources/simple_cacm_corpus.json'
+        with open(self.test_file) as f:
+            for line in f:
+                line = json.loads(line)
+                self.docids.append(line['id'])
+                self.texts.append(line['contents'])
+    
+    def assertIsFile(self, path):
+        if not pl.Path(path).resolve().is_file():
+            raise AssertionError("File does not exist: %s" % str(path))
+
+    def prepare_encoded_collection(self):
+        cache_dir = get_cache_home()
+        encoded_corpus_dir = f'{cache_dir}/temp_index'
+        cmd = f'python -m pyserini.encode \
+                input   --corpus {self.test_file} \
+                        --fields text \
+                output  --embeddings {encoded_corpus_dir} \
+                        --to-faiss \
+                encoder --encoder castorini/tct_colbert-v2-hnp-msmarco \
+                        --fields text \
+                        --batch 1 \
+                        --device cpu'
+        status = os.system(cmd)
+        self.assertEqual(status, 0)
+        self.assertIsFile(os.path.join(encoded_corpus_dir, 'docid'))
+        self.assertIsFile(os.path.join(encoded_corpus_dir, 'index'))
+        return encoded_corpus_dir
+
+    # def test_faiss_hnsw_pq(self):
```

Comment:
still working on this one, seems it's having problem with train() using only 3 data points.

Reply:
no need to try too hard... if it doesn't work, just axe it.